Fix build issues for SA_MODE:
```
../sa/libcubridsa.so.12.0: undefined reference to `is_active_transaction_server()'
../sa/libcubridsa.so.12.0: undefined reference to `is_passive_server()'
```
Define the functions in SA_MODE; stand-alone mode is considered an active transaction server.

--- a/src/base/server_type.cpp
+++ b/src/base/server_type.cpp
@@ -128,13 +128,13 @@ int init_server_type (const char *db_name)
     {
       assert (ts_Gl == nullptr);
 
-      if (is_active_transaction_server())
+      if (is_active_transaction_server ())
 	{
-	  ts_Gl.reset (new active_tran_server());
+	  ts_Gl.reset (new active_tran_server ());
 	}
-      else if (is_passive_transaction_server())
+      else if (is_passive_transaction_server ())
 	{
-	  ts_Gl.reset (new passive_tran_server());
+	  ts_Gl.reset (new passive_tran_server ());
 	}
       else
 	{
@@ -261,6 +261,31 @@ void finalize_server_type ()
 bool is_tran_server_with_remote_storage ()
 {
   return false;
+}
+
+bool is_active_transaction_server ()
+{
+  return true;
+}
+
+bool is_page_server ()
+{
+  return false;
+}
+
+bool is_passive_transaction_server ()
+{
+  return false;
+}
+
+bool is_passive_server ()
+{
+  return false;
+}
+
+bool is_transaction_server ()
+{
+  return true;
 }
 
 #endif // !SERVER_MODE = SA_MODE

--- a/src/server/passive_tran_server.cpp
+++ b/src/server/passive_tran_server.cpp
@@ -19,9 +19,15 @@
 #include "passive_tran_server.hpp"
 
 bool
+passive_tran_server::uses_remote_storage () const
+{
+  return true;
+}
+
+bool
 passive_tran_server::get_remote_storage_config ()
 {
-  return false;
+  return true;
 }
 
 void

--- a/src/server/passive_tran_server.hpp
+++ b/src/server/passive_tran_server.hpp
@@ -28,6 +28,7 @@ class passive_tran_server : public tran_server
     {
     }
 
+    bool uses_remote_storage () const final override;
   private:
     void on_boot () final override;
     bool get_remote_storage_config () final override;

--- a/src/storage/slotted_page.c
+++ b/src/storage/slotted_page.c
@@ -5276,5 +5276,7 @@ spage_need_compact (THREAD_ENTRY * thread_p, PAGE_PTR page_p)
 bool
 spage_is_save_space_disabled ()
 {
-  return log_is_in_crash_recovery () || is_passive_server ();
+  const bool during_crash_recovery = log_is_in_crash_recovery ();
+  const bool in_passive_server = is_passive_server ();
+  return during_crash_recovery || in_passive_server;
 }

--- a/src/transaction/boot_sr.c
+++ b/src/transaction/boot_sr.c
@@ -2659,7 +2659,7 @@ boot_restart_server (THREAD_ENTRY * thread_p, bool print_restart, const char *db
 #endif
 
 #if defined (SERVER_MODE)
-  if (get_server_type () == SERVER_TYPE_TRANSACTION)
+  if (is_active_transaction_server ())
     {
       /* set number of hosts */
       css_set_ha_num_of_hosts (db->num_hosts);

--- a/src/transaction/log_manager.c
+++ b/src/transaction/log_manager.c
@@ -1414,7 +1414,7 @@ log_initialize_internal (THREAD_ENTRY * thread_p, const char *db_fullname, const
        */
       log_recovery (thread_p, is_media_crash, stopat);
     }
-  else if (is_tran_server_with_remote_storage ()
+  else if (is_tran_server_with_remote_storage () && is_active_transaction_server ()
 	   && init_emergency == false && (log_Gl.m_metainfo.get_clean_shutdown () == false || is_media_crash == true))
     {
       log_recovery_finish_transactions (thread_p);

--- a/unit_tests/log/test_main_prior_list_serialize.cpp
+++ b/unit_tests/log/test_main_prior_list_serialize.cpp
@@ -330,6 +330,12 @@ get_server_type ()
   return SERVER_TYPE_TRANSACTION;
 }
 
+bool
+is_active_transaction_server ()
+{
+  return true;
+}
+
 PGLENGTH
 db_log_page_size ()
 {

